### PR TITLE
chore(nargo): Toggle brillig debug trace from CLI

### PR DESCRIPTION
**Pull request by vezenovm**
# Description

## Problem\*

Reference this comment: https://github.com/noir-lang/noir/pull/1755#pullrequestreview-1487752018

## Summary\*

Currently we have a hardcoded constant in the code for whether we display a Brillig debug trace. It is currently set to true which means anytime that a Brillig function is used in a program the trace is printed as well. Now that println has been set to use Brillig it means that every time println is used in a program the full brillig debug trace will be first outputted. 

I have added a flag `--show-brillig` that toggles this feature. I had to thread it further into the Brillig gen module than I would have liked due to how the current Brillig printer is implemented. The Brillig debug trace is more high-level and is more of a Brillig codegen printer than a full Brillig printer. For example, when allocating an array the IR printer will print this:
```
            "  ALLOCATE_ARRAY {} SIZE {}",
            pointer_register,
            size_register
```
but the actual Brillig generate to allocate an array is a MOV (to move the stack pointer to the pointer register), and a BinaryIntOp::Add (to increment the stack pointer). This high-level printing occurs for several instructions. I wanted originally to incorporate usage of the `Display` trait, but that will be a larger refactor is we want to keep this higher-level printer.

The debug_logs test now just prints this:
```
"*** println ***"
"0x05"
["0x05","0x0a"]
{"foo":"0x0f","my_struct":{"x":"0x0a","y":"0x05"}}
```
Rather than:
```
f4:
  CONST R6 = 198
  ALLOCATE_ARRAY R5 SIZE R6
  MOV R5, Stack
  Stack = Stack + R6
  .....
  like 900 lines of Brillig IR
  ......
  STORE *R5 = R12
  R5 = R5 + R6
  CONST R8 = 34
  STORE *R5 = R8
  R5 = R5 + R6
  CONST R26 = 125
  STORE *R5 = R26
  R5 = R5 + R6
  CONST R26 = 125
  STORE *R5 = R26
  R5 = R5 + R6
  FOREIGN_CALL println (R2[0..2], R3[0..51]) => 
  // return ;
  STOP
"*** println ***"
"0x05"
["0x05","0x0a"]
{"foo":"0x0f","my_struct":{"x":"0x0a","y":"0x05"}}
```

## Documentation

- [] This PR requires documentation updates when merged.

  

  - [] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: crates/noirc_evaluator/src/brillig/brillig_gen/brillig_slice_ops.rs [hunk 1]
```diff
@@ -260,7 +260,7 @@ mod tests {
         arguments: Vec<BrilligParameter>,
         returns: Vec<BrilligParameter>,
     ) -> BrilligContext {
-        let mut context = BrilligContext::new(arguments, returns);
+        let mut context = BrilligContext::new(arguments, returns, true);
```
Comment (jfecher) [inline crates/noirc_evaluator/src/brillig/brillig_gen/brillig_slice_ops.rs:263]: Is `create_context` only for testing?

File: crates/noirc_evaluator/src/brillig/brillig_ir/debug_show.rs [hunk 2]
```diff
@@ -130,215 +127,313 @@ impl<T: DebugToString> DebugToString for [T] {
...
+            false => ()
         }
     };
-    ( $format_message:expr, $( $x:expr ),* ) => {
-        if ENABLE_DEBUG_TRACE {
-            println!($format_message, $( $x.debug_to_string(), )*)
+    ( $enable_debug:expr, $format_message:expr, $( $x:expr ),* ) => {
+        match $enable_debug {
+            true => println!($format_message, $( $x.debug_to_string(), )*),
+            false => ()
```
Comment (jfecher) [inline crates/noirc_evaluator/src/brillig/brillig_ir/debug_show.rs:139]: These should probably be `if`s instead of matches.

**Review by jfecher — APPROVED** [review]

File: crates/noirc_evaluator/src/brillig/brillig_gen/brillig_slice_ops.rs (hunk 1 above)
Comment (vezenovm) [reply crates/noirc_evaluator/src/brillig/brillig_gen/brillig_slice_ops.rs:263]: Yes only for testing

Comment (jfecher) [reply crates/noirc_evaluator/src/brillig/brillig_gen/brillig_slice_ops.rs:263]: We should probably add `#[cfg(test)]` to it then or put it in a test module.

File: crates/noirc_evaluator/src/brillig/brillig_ir/debug_show.rs (hunk 2 above)
Comment (vezenovm) [reply crates/noirc_evaluator/src/brillig/brillig_ir/debug_show.rs]: I switched to if statement and it dismissed your review if you could approve once more @jfecher

File: crates/noirc_evaluator/src/brillig/brillig_gen/brillig_slice_ops.rs (hunk 1 above)
Comment (vezenovm) [reply crates/noirc_evaluator/src/brillig/brillig_gen/brillig_slice_ops.rs:263]: Ok I'll do that

Comment (vezenovm) [reply crates/noirc_evaluator/src/brillig/brillig_gen/brillig_slice_ops.rs:263]: It is actually within a tests module

**Review by jfecher — APPROVED** [review]

**Pull request merged**